### ISSUE TITLE
Look at bloomberg/PowerfulSeal for testing

### ISSUE THREAD

**Issue by lanzafame** [ipfs/kubernetes-ipfs#49]
@hsanjuan @ZenGround0 Do either of you see any value in using a tool like [PowerfulSeal](https://github.com/bloomberg/powerfulseal)? It is sort-of ChaosMonkey for Kubernetes.

**Comment by ZenGround0** [comment]
Yes!  I definitely see value in using something like this.  In the past I hacked together some very rough tooling to accomplish a simple version of this, see the [process running in the cluster testing container](https://github.com/ipfs/ipfs-cluster/blob/master/docker/random-killer.sh) and the [k8s test making use of it](https://github.com/ipfs/kubernetes-ipfs/blob/master/ipfs-cluster/tests/0012-random_kill.yml).

Doing something more principled would be very good and at a glance PowerfulSeal would be a good candidate for this.  When it comes to tests that bring processes up and down one of the trickier things to keep in mind is persisting state between periods of unavailability.  I built out some hacky things to accomplish a version of this that gets the job done but they're inelegant and difficult to use and extend.  It would be great to re-examine this part of k8s-ipfs at the moment the random failures behavior is re-evaluated.
